### PR TITLE
[go_router_builder] Fixes incorrect separator at location path on Windows.

### PR DESCRIPTION
**Pull request by moko256**
The `go_router_builder` is using `path.joinAll` to concat `route` and create `location`, but it uses backslash as path separators on Windows.
This PR will replace that with `path.url.joinAll`, which uses slash. The builder will be able to generate slash-separated location on Windows, same on the other platforms.

I didn't know how to write test in this case. The build runner will generate correct `main.g.dart` in example project on Windows.

This PR will fix flutter/flutter#102710.

No changes in `flutter/tests`

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by chunhtai — CHANGES_REQUESTED** [review]
it looks like the change log is out of date, can you rebase off the main branch and fix the conflict?

The change makes sense, but we need a test in order to merge this pr. Can you write a example that cover this use case in example/ folder?

**Comment by moko256** [comment]
I rebased it and pushed it.

The main.dart in example already covers this, which is included as a reproducible example in #102710.

https://github.com/flutter/packages/blob/2c9dbda11a7e3b9dac6ef56e18cbfec3f2ca61f9/packages/go_router_builder/example/lib/main.dart#L65-L80

**Comment by chunhtai** [comment]
> I rebased it and pushed it.
> 
> The main.dart in example already covers this. These line is included as a reproducible example in [flutter/flutter#102710](https://github.com/flutter/flutter/issues/102710).
> 
> https://github.com/flutter/packages/blob/2c9dbda11a7e3b9dac6ef56e18cbfec3f2ca61f9/packages/go_router_builder/example/lib/main.dart#L65-L80

but there is no file main.g.dart does not change after this change. is it needed to be run in windows host?

**Comment by moko256** [comment]
> > I rebased it and pushed it.
> > The main.dart in example already covers this. These line is included as a reproducible example in [flutter/flutter#102710](https://github.com/flutter/flutter/issues/102710).
> > https://github.com/flutter/packages/blob/2c9dbda11a7e3b9dac6ef56e18cbfec3f2ca61f9/packages/go_router_builder/example/lib/main.dart#L65-L80
> 
> but the file main.g.dart does not change after this change. is it needed to be run in windows host?

Yes, and to see generated result you have to re-generate main.g.dart with `flutter pub run build_runner build` on Windows.
The generated main.g.dart with a version before this PR on Windows differs from committed one.

**Comment by chunhtai** [comment]
sounds good, I am working on another pr to run test on windows platform, so test wise, this pr should be good

based on the CI

�[31mThe following packages had errors:�[0m
�[31m  go_router_builder:
    Missing version change�[0m
�[31mSee above for full details.�[0m

you need to bump the package version.

File: packages/go_router_builder/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ## 1.0.2
 
+- Fixes incorrect separator at location path on Windows. [#102710](https://github.com/flutter/flutter/issues/102710)
```
Comment (chunhtai) [inline packages/go_router_builder/CHANGELOG.md:3]: Can you add the new change to 1.0.3 and bump the package yaml to 1.0.3 as well?

**Review by chunhtai — APPROVED** [review]
LGTM

File: packages/go_router_builder/CHANGELOG.md (hunk 1 above)
Comment (moko256) [reply packages/go_router_builder/CHANGELOG.md:3]: Done. (I missed v1.0.2 release when rebasing)

File: packages/go_router_builder/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
+## 1.0.3
+
+- Fixes incorrect separator at location path on Windows. [#102710](https://github.com/flutter/flutter/issues/102710)
+
 ## 1.0.2
 
 - Changes the parameter name of the auto-generated `go` method from `buildContext` to `context`.
+- Fixes integration tests. [#102799](https://github.com/flutter/flutter/issues/102799).
```
Comment (kevmoo) [inline packages/go_router_builder/CHANGELOG.md:8]: drop this line.
```suggestion
```

**Review by kevmoo — APPROVED** [review]

File: packages/go_router_builder/CHANGELOG.md (hunk 2 above)
Comment (moko256) [reply packages/go_router_builder/CHANGELOG.md]: Done

**Review by kevmoo — APPROVED** [review]
Great find! Thanks!

**Pull request merged**